gpu: document required system-probe.yaml flag and gpu.enabled for non-containerized Linux hosts (#23853)
DataDog/integrations-core: gpu/README.md

#### Host

> GPU monitoring requires configuration in **both** `/etc/datadog-agent/datadog.yaml` and `/etc/datadog-agent/system-probe.yaml`. Configuring only one of these files results in incomplete metrics collection.

Reviewer: ```suggestion
GPU monitoring requires configuration in both `/etc/datadog-agent/datadog.yaml` and `/etc/datadog-agent/system-probe.yaml`. Configuring only one of these files results in incomplete metrics collection.
```

> **Step 1**: Add the following parameters to `/etc/datadog-agent/datadog.yaml`:

Reviewer: ```suggestion
1. Add the following parameters to `/etc/datadog-agent/datadog.yaml`:
```

```yaml
gpu:
  enabled: true
collect_gpu_tags: true
enable_nvml_detection: true
```

> **Step 2**: Add the following parameters to `/etc/datadog-agent/system-probe.yaml`. This flag loads the eBPF module responsible for per-process GPU metrics and is required even for non-containerized hosts:

Reviewer: ```suggestion
2. Add the following parameter to `/etc/datadog-agent/system-probe.yaml`. This flag loads the eBPF module responsible for per-process GPU metrics and is required even for non-containerized hosts:
```

```yaml
gpu_monitoring:
  enabled: true
```

> **Step 3**: Restart both the Agent and system-probe:

Reviewer: ```suggestion
3. Restart both the Agent and system-probe:
```

```shell
sudo systemctl restart datadog-agent
sudo systemctl restart datadog-agent-sysprobe
```

> The check in the Agent configuration file is also enabled by default whenever NVIDIA GPUs and their drivers are detected in the system, as long as the `enable_nvml_detection` parameter is set to `true`. However, it can also be configured manually following these steps:

Reviewer: ```suggestion
The check in the Agent configuration file is enabled by default whenever NVIDIA GPUs and their drivers are detected in the system, as long as the `enable_nvml_detection` parameter is set to `true`. The check can also be configured manually following these steps:
```
